### ISSUE TITLE
starfox crashes before starting second level (sector y)

### ISSUE THREAD

**Issue by scottjg** [Provenance-Emu/Provenance#1035]
I ran star fox 64 with the latest build on my non-4k appletv. i completed the first level using the "secret" path where you follow Falco through the waterfall. after the mission, the star map is displayed and I pick the next level, sector y, and the emulator immediately crashes (i played through it twice to make sure it wasn't a one-off)

### Expected:  
not to crash

### Actual:  
it crashed

### Reproduce:  
1. launch starfox 64
2. play the first level through the secret path through the waterfall
3. try to start the next level

# 
#### NOTES  
i captured the logs from the crash. lmk if there's anything i can do to help debug.

```
Date/Time:       2018-12-09 21:59:55 -0800
End time:        2018-12-09 22:00:01 -0800
OS Version:      Apple TVOS 12.1 (Build 16J602)
Architecture:    arm64
Report Version:  27
Incident Identifier: 91E8E38D-DB65-4FD7-AD26-A438BEC0CC28

Data Source:     Microstackshots
Shared Cache:    0x405c000 38BE9457-582B-3E87-9F2E-E1DE007AC19E

Command:         Provenance
Path:            /private/var/containers/Bundle/Application/6C00578B-3A03-4165-9FD4-D5A0C4B82997/Provenance.app/Provenance
Identifier:      com.scottjg.provenance
Version:         1.5 (2706)
Beta Identifier: 3A7E3C33-140C-46D5-A477-62C8E293E70B
PID:             5428

Event:           wakeups
Action taken:    none
Wakeups:         45001 wakeups over the last 251 seconds (180 wakeups per second average), exceeding limit of 150 wakeups per second over 300 seconds
Wakeups limit:   45000
Limit duration:  300s
Wakeups caused:  45001
Duration:        6.03s
Steps:           7

Hardware model:  AppleTV5,3
Active cpus:     2
Boot args:        


Heaviest stack for the target process:
  6  ??? (libsystem_pthread.dylib + 9112) [0x184b27398]
  6  ??? (libsystem_pthread.dylib + 9272) [0x184b27438]
  1  ??? (<F16EF5B7-D9BD-38D1-AE0D-4761D9CADAB5> + 165604) [0x10c8e06e4]


Powerstats for:  Provenance [5428]
UUID:            F52183EE-9F4D-3862-92CA-19875DFC5BA8
Path:            /private/var/containers/Bundle/Application/6C00578B-3A03-4165-9FD4-D5A0C4B82997/Provenance.app/Provenance
Architecture:    arm64
Parent:          UNKNOWN [1]
UID:             501
Footprint:       1108.56 MB
Start time:      2018-12-09 21:59:55 -0800
End time:        2018-12-09 22:00:01 -0800
Num samples:     7 (100%)
CPU Time:        3.670s
Primary state:   5 samples Frontmost App, Non-Suppressed, User mode, Effective Thread QoS Default, Requested Thread QoS Default, Override Thread QoS Unspecified
User Activity:   0 samples Idle, 0 samples Active, 7 samples Unknown
Power Source:    0 samples on Battery, 0 samples on AC, 7 samples Unknown
  6  _pthread_start + 47 (libsystem_pthread.dylib + 9112) [0x184b27398]
    6  _pthread_body + 127 (libsystem_pthread.dylib + 9272) [0x184b27438]
      1  vi_vertical_interrupt_event (vi_controller.c:153 in PVMupen64Plus + 195572) [0x10c8e7bf4]
        1  VI_UpdateScreen (VI.cpp:178) (VI.cpp:178 in PVMupen64PlusVideoGlideN64 + 81628) [0x1053bfedc]
          1  FrameBufferList::renderBuffer (FrameBuffer.cpp:1481) (FrameBuffer.cpp:1481 in PVMupen64PlusVideoGlideN64 + 305764) [0x1053f6a64]
            1  -[MupenGameCore swapBuffers] (MupenGameCore.m:1223 in PVMupen64Plus + 142036) [0x10c8daad4]
              1  -[PVGLViewController didRenderFrameOnAlternateThread] (PVGLViewController.m:677 in Provenance + 39968) [0x1008c9c20]
                1  ??? (AGXGLDriver + 152264) [0x19deca2c8]
                  1  gpusSubmitDataBuffers + 143 (libGPUSupportMercury.dylib + 20196) [0x19c2c9ee4]
                    1  IOAccelContextSubmitDataBuffersExt2 + 211 (IOAccelerator + 17120) [0x18709d2e0]
                      1  IOConnectCallStructMethod + 51 (IOKit + 26800) [0x18517f8b0]
                        1  IOConnectCallMethod + 231 (IOKit + 26432) [0x18517f740]
                          1  io_connect_method + 415 (IOKit + 448504) [0x1851e67f8]
                            1  mach_msg_trap + 8 (libsystem_kernel.dylib + 102096) [0x184a9ded0]
                              1  <Kernel mode>
      1  r4300_write_aligned_word (r4300_core.c:343 in PVMupen64Plus + 165604) [0x10c8e06e4]
      1  write_rsp_regs (rsp_core.c:278 in PVMupen64Plus + 155348) [0x10c8dded4]
        1  DoRspCycles (plugin.c:439 in PVMupen64PlusRspHLE + 54572) [0x10c6a952c]
          1  send_dlist_to_gfx_plugin (hle.c:172 in PVMupen64PlusRspHLE + 37012) [0x10c6a5094]
            1  RSP_ProcessDList (RSP.cpp:178) (RSP.cpp:178 in PVMupen64PlusVideoGlideN64 + 103880) [0x1053c55c8]
              1  GraphicsDrawer::drawTriangles (GraphicsDrawer.cpp:757) (GraphicsDrawer.cpp:757 in PVMupen64PlusVideoGlideN64 + 430684) [0x10541525c]
                1  glDrawElements_IMM_ES2Exec + 247 (GLEngine + 398648) [0x19eaba538]
                  1  gleDoDrawDispatchCoreES2 + 639 (GLEngine + 559840) [0x19eae1ae0]
                    1  gldUpdateDispatch + 684 (AGXGLDriver + 142260) [0x19dec7bb4]
      1  read_rdram_dram (rdram.c:234 in PVMupen64Plus + 108564) [0x10c8d2814]
      1  cached_interp_MTC0 (mips_instructions.def:1179 in PVMupen64Plus + 82512) [0x10c8cc250]
      1  cached_interp_JR_OUT (mips_instructions.def:893 in PVMupen64Plus + 62292) [0x10c8c7354]
  1  start + 3 (libdyld.dylib + 7724) [0x184946e2c]
    1  main (PVAppDelegate.swift:19 in Provenance + 186388) [0x1008ed814]
      1  UIApplicationMain + 211 (UIKitCore + 8474596) [0x1a42e1fe4]
        1  GSEventRunModal + 99 (GraphicsServices + 46468) [0x187158584]
          1  CFRunLoopRunSpecific + 435 (CoreFoundation + 681492) [0x184ea3614]
            1  __CFRunLoopRun + 2139 (CoreFoundation + 684432) [0x184ea4190]
              1  __CFRunLoopDoSource1 + 439 (CoreFoundation + 705496) [0x184ea93d8]
                1  __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 55 (CoreFoundation + 707724) [0x184ea9c8c]
                  1  __CFMachPortPerform + 187 (CoreFoundation + 546016) [0x184e824e0]
                    1  IODispatchCalloutFromCFMessage + 487 (IOKit + 23444) [0x18517eb94]
                      1  CA::Display::DisplayLink::dispatch_items + 827 (QuartzCore + 80708) [0x1892c4b44]
                        1  -[GLKViewController _updateAndDraw] + 519 (GLKit + 158488) [0x19177fb18]
                          1  -[GLKView _display:] + 307 (GLKit + 153916) [0x19177e93c]
                            1  -[EAGLContext presentRenderbuffer:] + 79 (OpenGLES + 35468) [0x1881a5a8c]
                              1  gliPresentViewES_Exec + 187 (GLEngine + 939236) [0x19eb3e4e4]
                                1  ??? (AGXGLDriver + 152264) [0x19deca2c8]
                                  1  gpusSubmitDataBuffers + 143 (libGPUSupportMercury.dylib + 20196) [0x19c2c9ee4]
                                    1  IOAccelContextSubmitDataBuffersExt2 + 211 (IOAccelerator + 17120) [0x18709d2e0]
                                      1  IOConnectCallStructMethod + 51 (IOKit + 26800) [0x18517f8b0]
                                        1  IOConnectCallMethod + 231 (IOKit + 26432) [0x18517f740]
                                          1  io_connect_method + 415 (IOKit + 448504) [0x1851e67f8]
                                            1  mach_msg_trap + 8 (libsystem_kernel.dylib + 102096) [0x184a9ded0]
                                              1  <Kernel mode, Effective Thread QoS User Interactive, Requested Thread QoS User Interactive>

  Binary Images:
           0x1008c0000 -                ???  Provenance                 <F52183EE-9F4D-3862-92CA-19875DFC5BA8>  /private/var/containers/Bundle/Application/6C00578B-3A03-4165-9FD4-D5A0C4B82997/Provenance.app/Provenance
           0x1053ac000 -                ???  ???                        <13D642B7-76A1-3EF4-99E9-1B6F8C560CF9>  PVMupen64PlusVideoGlideN64
           0x10c69c000 -                ???  ???                        <FBCCB82B-5B04-378E-BAAD-0CFC84E756F5>  PVMupen64PlusRspHLE
           0x10c8b8000 -                ???  ???                        <F16EF5B7-D9BD-38D1-AE0D-4761D9CADAB5>  PVMupen64Plus
           0x184945000 -        0x184970fff  libdyld.dylib              <CDF19092-0A2D-3ADE-AEBF-65FC7A33718D>  /usr/lib/system/libdyld.dylib
           0x184a85000 -        0x184ab0fff  libsystem_kernel.dylib     <0C55DA5F-CB7D-3BF0-8904-D3935FAAA253>  /usr/lib/system/libsystem_kernel.dylib
           0x184b25000 -        0x184b30fff  libsystem_pthread.dylib    <E3E622E6-2FE8-30D6-8915-1FB177DF6558>  /usr/lib/system/libsystem_pthread.dylib
           0x184dfd000 -        0x185160fff  CoreFoundation             <7C0E2C1E-B591-3001-8AA8-9C3D2B4B3780>  /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation
           0x185179000 -        0x185208fff  IOKit                      <6B0242F8-B08E-3ED4-B832-7FE54F22464F>  /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
           0x187099000 -        0x1870a0fff  IOAccelerator              <7BF5F52A-87D3-3EDE-B55A-54BD7ABAA1C2>  /System/Library/PrivateFrameworks/IOAccelerator.framework/IOAccelerator
           0x18714d000 -        0x187160fff  GraphicsServices           <BC2F3FC4-0B97-3C46-AAF9-18727D0312AF>  /System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices
           0x18819d000 -        0x1881a8fff  OpenGLES                   <DFADAA7F-E884-3693-81EA-DCB98C391E54>  /System/Library/Frameworks/OpenGLES.framework/OpenGLES
           0x1892b1000 -        0x1894e4fff  QuartzCore                 <1E1DAE72-668E-342A-BF03-5830FD015BD4>  /System/Library/Frameworks/QuartzCore.framework/QuartzCore
           0x191759000 -        0x19178cfff  GLKit                      <ADEAF995-8955-3508-86D3-F082A9109B68>  /System/Library/Frameworks/GLKit.framework/GLKit
           0x19c2c5000 -        0x19c2d0fff  libGPUSupportMercury.dylib <A330DA9E-0C70-36D5-926C-BD1146066D00>  /System/Library/PrivateFrameworks/GPUSupport.framework/libGPUSupportMercury.dylib
           0x19dea5000 -        0x19df1cfff  AGXGLDriver                <9F69D40D-BB4E-3668-90BD-52AD81203052>  /System/Library/Extensions/AGXGLDriver.bundle/AGXGLDriver
           0x19ea59000 -        0x19eb44fff  GLEngine                   <7B2CADB5-247C-31AA-A231-0BD6B57740C0>  /System/Library/Frameworks/OpenGLES.framework/GLEngine.bundle/GLEngine
           0x1a3acd000 -        0x1a4a74fff  UIKitCore                  <249935DE-709D-3EAB-BDB7-DC52977DEB0D>  /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore
```

#

#### APP VERSION
- [ ] 1.5 *beta*
  - [x] latest build (840ae6d7dbcf469085895945e8f08d338e88df12)
- [ ] 1.4

#### APP SOURCE
- [x] Provenance-Emu GitHub  *&nbsp;(Official)*
- [ ] Other  *&nbsp;(Unofficial)*

#### INSTALLED BY 
- [ ] [Side-Loading](https://git.io/vpYzf)
- [x] [Building](https://git.io/vpYzv) *from Source*
  - [x] Xcode 9.3+ (10.1)
  - [ ] Xcode 9.2 *or lower*

#### PLATFORM
- [ ] iPhone
- [ ] iPad
- [x] Apple TV

#### iOS/tvOS VERSION
- [x] Latest (11.3)
- [ ] 10.x
- [ ] 9.x
- [ ] 8.x

**Comment by JoeMatt** [comment]
This is probably something in Mupen and or Gliden and not really in the domain of a bug in Provenance. We don't make a habit of debugging upstream project bugs.

Is this reproducible in an other iOS Gliden/Mupen apps, like retroarch for iOS? Or maybe even an OS X version?

I'll leave this open to track for a little bit but marking it won't fix since honestly no one is going to fix this from this project. There are compatibility flags for Mupen/Gliden that need to be exposed once per core / game configurations are done. There are some hard coded settings that aren't always  100% compatible with all games so it just is what it is until we expose the ability to change options

**Comment by JoeMatt** [comment]
closing due to age, many updates since. Reopen if persists in 2.1.0 or later
